### PR TITLE
Fix some test errors and deprecation warnings on 0.6

### PR DESCRIPTION
**Pull request by TotalVerb**
This fixes uses of `s.data` that prevent Compat tests from passing. In particular, we no longer check that `unsafe_wrap` aliases properly on sufficiently new versions of 0.6.

This also fixes several deprecation warnings from `@test_approx_eq`.

This does not fix the following remaining deprecation warnings:

- `floor`, `round`, `ceil`, etc. as vectorized functions
- `promote_eltype_op`

because both fixes will likely require (non-test) code changes. I have a solution to `promote_eltype_op` that I will open as a PR shortly.

### CODE REVIEW

**Comment by TotalVerb** [comment]
Closing and reopening to restart Appveyor.

**Comment by stevengj** [comment]
Since `unsafe_wrap(String, ...)` has disappeared in 0.6, should we drop it from Compat?

**Comment by TotalVerb** [comment]
For as long as we continue to support 0.4, it would break packages that have not yet upgraded to 0.6 if we deprecated or dropped `unsafe_wrap(String, ...)`. However I do agree that we should document that this feature is no longer available in 0.6 or later.

File: test/runtests.jl [hunk 1]
```diff
@@ -1304,19 +1304,29 @@ io = IOBuffer()
...
+    end
     new_str = unsafe_string(ptr)
     cstr = convert(Cstring, ptr)
     new_str2 = unsafe_string(cstr)
     @test wrapped_str == "test"
     @test new_str == "test"
     @test new_str2 == "test"
-    @test ptr == pointer(wrapped_str)  # Test proper pointer aliasing behavior
+    if VERSION < v"0.6.0-dev.1988"
+        # Test proper pointer aliasing behavior, which is of course not
```
Comment (tkelman) [inline test/runtests.jl]: "of course" is debatable 😉

Comment (TotalVerb) [inline test/runtests.jl]: Removed this part.

**Comment by TotalVerb** [comment]
Is this good to merge? I'll handle the remaining deprecations and documentation issues later but it's useful to get Travis working again.

**Pull request merged**